cordyceps/src/list.rs

```diff
@@ -897,6 +922,44 @@ impl<T: Linked<Links<T>> + ?Sized> List<T> {
         }
     }
 
+    /// Returns an iterator over the items in this list, by pointer.
+    ///
+    /// ## Safety
+    ///
+    /// Although this method is safe, care must be taken with the items
+    /// yielded by the returned iterator.
+    ///
+    /// This iterator returns `NonNull<T>` of the individual node elements,
+    /// rather than references. This is done to allow "type punning" of nodes,
+    /// where the creation of a reference could restrict the provenance of the
+    /// pointed-to item. This is relevant if your nodes are of a common header
+    /// type, but may have different "body" types you'd like to cast to. This is similar
+    /// to how executors like `maitake` or `tokio` store `Task`s: with a common
+    /// `TaskHeader` but containing varying futured in their bodies. If this is
```

Comment:
maybe worth linking to one of those examples? although maybe it's sprayed too far across those codebases to be easily referenceable. we could also try to turn some of your nice test into an example we could reference here.

Reply:
We could, though those use different structures (mpscqueue, transferstack), so it might not be the cognitive load to follow.

the test could be useful, it requires a lot of moving bits as a minimal example tho.